### PR TITLE
[RAINCATCH-507] Update to handle sync remote events

### PR DESCRIPTION
**Pull request by JameelB**
# What
Allow users to view the current sync status of a workflow on-device, notifying them if there are any errors that occurred during sync to the remote data store. 

# How
The current sync status is included in the workflow object received from cloud. In the workflow list template, a sync error icon and message are added and shown when a remote update failed occurs. 

*UPDATE*
The sync module only adds the syncStatus property to the workflow item if the sync event failed. Instead of checking for the type of the sync status, check if sync status exist instead. 

# JIRA
https://issues.jboss.org/browse/RAINCATCH-507

### CODE REVIEW

File: dist/workflow-list.tpl.html.js [hunk 1]
```diff
@@ -24,7 +24,11 @@ ngModule.run(['$templateCache', function ($templateCache) {
     '  <md-list-item ng-repeat="workflow in ctrl.workflows" ng-click="ctrl.selectWorkflow($event, workflow)" ng-class="{active: selected.id === workflow.id}" >\n' +
     '    <div class="md-list-item-text">\n' +
     '      <p>\n' +
-    '        {{workflow.title}}\n' +
+    '        {{workflow.title}} <br/>\n' +
+    '        <md-icon ng-show="workflow._syncStatus.type === \'failed\'" md-font-set="material-icons" class="material-icons sync-error-icon">\n' +
```
Comment (ghost) [inline dist/workflow-list.tpl.html.js]: `ng-if` drirective with workflow._syncStatusFaild param check? Think better create some scope attribute `$scope.syncStatusFailed` and play with it rather compare values directly in template

**Review by vchepeli — COMMENTED** [review]
LGTM

**Comment by JameelB** [comment]
@vchepeli updated according to review

**Review by witmicko — APPROVED** [review]
LGTM

**Pull request merged**